src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const cloneOfStates = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete cloneState[key];
+        }
+        break;
 
+      case 'clear':
+        for (const key of Object.keys(cloneState)) {
+          delete cloneState[key];
+        }
+        break;
+
+      default:
```

Comment:
it is a good practice when you add a default behavior for switch statement. But don't let it be empty. If you have no default case you can just throw an Error with text like "unexpected action"